test/specs/commonTests.js

```diff
@@ -408,10 +408,13 @@ export const implementsAlignedProp = (Component, requiredProps = {}) => {
 export const implementsIconProp = (Component, requiredProps = {}) => {
   const iconClass = faker.hacker.noun()
   const assertValid = (wrapper) => {
-    wrapper.should.have.className('icon')
-    wrapper.should.have.descendants('Icon')
-    wrapper.find('Icon')
-      .should.have.className(iconClass)
+    it('has an Icon descendant', () => {
+      wrapper.should.contain(<Icon name={iconClass} />)
+    })
+
+    it('has Icon descendant with icon class name', () => {
+      wrapper.find('Icon').should.have.className(iconClass)
+    })
```

Comment:
Now that we have the `name` prop, we can leave the `className` test to the Icon tests.  Let's just leave the first test that asserts there is an Icon descendant with the name prop set to `iconClass`.  We can remove this test for the `className`.

This should also remove the requirement to `mount` in the tests below this.  Since, the Icon no longer has to render down to markup with a className.  It only needs to be a composite component with the `name` prop set.